packages/gdl-frontend/apollo/initApollo.js

```diff
@@ -69,31 +79,39 @@ function create(initialState, { getToken }) {
   });
 
   return new ApolloClient({
+    fetch,
     connectToDevTools: process.browser,
-    ssrMode: !process.browser, // Disables forceFetch on the server (so queries are only run once)
+    ssrMode: !Boolean(process.browser), // Disables forceFetch on the server (so queries are only run once)
     link: authLink
+      .concat(retry)
       .concat(errorLink)
       .concat(timeoutLink)
       .concat(httpLink),
     cache: cache.restore(initialState || {}),
-    fetch
+    ssrForceFetchDelay: !Boolean(process.browser)
   });
 }
 
-const isCookiesEnabled = () => {
-  let cookieEnabled = typeof window !== 'undefined' && navigator.cookieEnabled;
-
-  if (!cookieEnabled) {
-    document.cookie = 'testcookie';
-    cookieEnabled = document.cookie.indexOf('testcookie') !== -1;
-  }
-  return cookieEnabled;
-};
+/**
+ * To make apollo work with offline, the queries in apollo cache needs to be persisted.
+ * window.__AOLLO_STATE__ provides a fresh apollo cache for every initial page visit
+ * and it accumulates the apollo responses for each new client.
+ */
 
-export async function loadCache() {
+export async function persistAndPopulateCache() {
   if (!!process.browser && isCookiesEnabled()) {
-    return await persistCache({ cache, storage: localForage });
-  } else return null;
+    await localForage.setItem(
+      'apollo-cache-persist',
+      JSON.stringify(window.__APOLLO_STATE__)
+    );
+
+    await persistCache({ cache, storage: localForage });
+    // Since window.__APOLLO_STATE__ accumulates only current apollo responses,
+    // to make offline work the offline queries needs to be repopulated.
```

Comment:
Kanskje reformulere til feks:
> Since window.__APOLLO_STATE__ accumulates only current apollo responses all offline queries needs to be repopulated in order to make offline browsing/reading of books possible.